src/main/kotlin/org/rust/ide/formatter/RustAlignmentStrategy.kt

```diff
@@ -0,0 +1,42 @@
+package org.rust.ide.formatter
+
+import com.intellij.formatting.Alignment
+import com.intellij.lang.ASTNode
+
+/**
+ * Mimics [com.intellij.formatting.alignment.AlignmentStrategy], but offers more flexible API
+ */
+interface RustAlignmentStrategy {
```

Comment:
Yep, builder DSL looks nice. 

Can you please elaborate why exactly you need lazyness? Is it because you set the alignment after the block is crated?  Will [this](https://github.com/intellij-rust/intellij-rust/pull/413#discussion_r64387669) help?

The laziness make me feel a little uneasy.